src/components/RequestToBeProvider.svelte

```diff
@@ -1,73 +1,100 @@
-      return;
+  // a callback for when a transaction hits a final state
+  let requestToBeProviderTxnFinished = async (succeeded: boolean) => {
+    if (succeeded) {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const msaInfo: MsaInfo = await getMsaInfo($dotApi.api!, $user.address);
+      $user.providerName = providerNameToHuman(msaInfo.providerName);
+      $user.isProvider = msaInfo.isProvider;
+      isInProgress = false;
+      setTimeout(() => {
+        pageContent.dashboard();
+      }, 1500);
     }
-    if (!localDotApi) {
-      alert('please reconnect to an endpoint');
-      return;
+  };
+
+  const checkIsFinished = async () => {
+    if (recentActivityItem && recentActivityItem.txnStatus !== TxnStatus.LOADING) {
+      await requestToBeProviderTxnFinished(recentActivityItem.txnStatus === TxnStatus.SUCCESS);
+      if (recentActivityItem.txnStatus === TxnStatus.FAILURE) isInProgress = false;
     }
-    await submitRequestToBeProvider(localDotApi.api as ApiPromise, await getExtension($user), $user, newProviderName);
-    isInProgress = false;
   };
 
   $: {
-    if (isInProgress) {
-      recentActivityItem = $activityLog[0];
-    }
+    recentActivityItem = $activityLog.find((value) => value.txnId === recentTxnId);
+    checkIsFinished();
   }
+
+  const doProposeToBeProvider = async (_evt: Event) => {
+    if (!$dotApi.api) {
+      alert('please reconnect to an endpoint');
+      return;
+    }
+    isInProgress = true;
+    recentTxnId = await submitRequestToBeProvider($dotApi.api, await getExtension($user), $user, newProviderName);
+    recentActivityItem = $activityLog.find((value) => value.txnId === recentTxnId);
+  };
 </script>
 
-<div id="request-to-be-provider" class="action-card basis-1/2">
-  <h2>Request to Be a Provider</h2>
-  <h3>What is a Provider?</h3>
-  <p>A Provider is an MSA holder on Frequency with special permissions.</p>
-  <ol>
-    <li>They can pay for transactions with Capacity as well as Frequency token.</li>
-    <li>They can be permitted to post certain transactions on another MSA's behalf, also known as delegation.</li>
-    <li>An MSA can stake token to generate Capacity, and designate a Provider to receive that Capacity.</li>
-  </ol>
-  <p>Anyone with an MSA ID on Frequency's Mainnet who wants to become a Provider must follow this process:</p>
-  <ol>
-    <li>Submit an on-chain transaction to request be become a provider by filling in and submitting the form below.</li>
-    <li>
-      <a href={mailTo} class="underline"> Contact the Frequency Council </a>
-      and inform them that you have requested to become a Provider, and provide them with your MSA Id.
-    </li>
-  </ol>
-  <form>
-    <label for="providerNameRtB">Provider name</label>
-    <input id="providerNameRtB" placeholder="Short name" maxlength="16" bind:value={newProviderName} />
+<div id="request-to-be-provider" class="column">
+  <form class="column">
```

Comment:
This could be a separate issue, but I think we need to adjust one thing at the completion step.

It should remove the form on success. It looks like you can still do something:

![image](https://github.com/user-attachments/assets/a1f6e182-c055-4540-8ad7-4164b4e4da23)

Just hiding the form is good enough for now I think:

<img width="624" alt="image" src="https://github.com/user-attachments/assets/13d23036-99e7-4274-8909-966267dd5879" />